src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -462,12 +506,15 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
             return;
         }
 
-        for (const key of content.keys) {
-            if (!key.key || !isNumber(key.index)) {
-                logger.warn(`Received m.call.encryption_keys with invalid entry: callId=${callId}`);
-                continue;
-            }
+        if (userId === this.client.getUserId() && deviceId === this.client.getDeviceId()) {
+            // We store our own sender key in the same set along with keys from others, so it's
+            // important we don't allow our own keys to be set by one of these events (apart from
+            // the fact that we don't need it anyway because we already know our own keys).
+            logger.info("Ignoring our own keys event");
+            return;
+        }
 
+        for (const key of content.keys) {
             const encryptionKey = key.key;
             const encryptionKeyIndex = key.index;
```

Comment:
These accesses are without a guard that checks whether `key` is null or undefined